runtime/src/bank.rs

```diff
@@ -5131,6 +5135,84 @@ pub(crate) mod tests {
         bank
     }
 
+    fn assert_capitalization_diff(bank: &Bank, updater: impl Fn(), asserter: impl Fn(u64, u64)) {
+        let old = bank.capitalization();
+        updater();
+        let new = bank.capitalization();
+        asserter(old, new);
+        assert_eq!(bank.capitalization(), bank.calculate_capitalization());
+    }
+
+    #[test]
+    fn test_store_account_and_update_capitalization_missing() {
+        let (genesis_config, _mint_keypair) = create_genesis_config(0);
+        let bank = Bank::new(&genesis_config);
+        let pubkey = solana_sdk::pubkey::new_rand();
+
+        let some_lamports = 400;
+        let account = Account::new(some_lamports, 0, &system_program::id());
+
+        assert_capitalization_diff(
+            &bank,
+            || bank.store_account_and_update_capitalization(&pubkey, &account),
+            |old, new| assert_eq!(old + some_lamports, new),
+        );
+        assert_eq!(account, bank.get_account(&pubkey).unwrap());
+    }
+
+    #[test]
+    fn test_store_account_and_update_capitalization_increased() {
+        let old_lamports = 400;
+        let (genesis_config, mint_keypair) = create_genesis_config(old_lamports);
+        let bank = Bank::new(&genesis_config);
+        let pubkey = mint_keypair.pubkey();
+
+        let new_lamports = 500;
+        let account = Account::new(new_lamports, 0, &system_program::id());
+
+        assert_capitalization_diff(
+            &bank,
+            || bank.store_account_and_update_capitalization(&pubkey, &account),
+            |old, new| assert_eq!(old + 100, new),
```

Comment:
trivia: Q: why do I this manual math hard code like this instead of `old + (new_lamports - old_lamports)` sometimes? Isn't a bad practice?

A: Well this is partly art of coding but these well nuanced hard coded math can catch really silly test failure. Say, I assigned same value to both `old_lamports` and `new_lamports` in a huste of refactoring by accident, then, `old + (new_lamports - old_lamports)` wouldn't detect that this test got broken meaning actually noting (including intended branch coverage in `store_account_and_update_capitalization`) is tested.

Also, note that these silly things could happen in the refactoring of `cfg(not(test))` code.